### PR TITLE
doc(database): Document off-cluster db configuration

### PR DESCRIPTION
**Pull request by krancour**
Complements https://github.com/deis/charts/pull/286

cc @bacongobbler

### CODE REVIEW

File: src/installing-workflow/configuring-postgres.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+> create user <deis username; typically "deis"> with password '<password>';
+> create database <database name; typically "deis"> with owner <deis username>;
+> \q
+```
+
+## Configuring Workflow
+
+The Helm Classic chart for Deis Workflow can be easily configured to connect the Workflow controller component to an off-cluster PostgreSQL database.
+
+* **Step 1:** If you haven't already fetched the Helm Classic chart, do so with `helmc fetch deis/workflow-v2.1.0`
```
Comment (bacongobbler) [inline src/installing-workflow/configuring-postgres.md]: if this is referring to v2.1.0, should we wait until we've shipped v2.1.0 to merge this?

Comment (krancour) [inline src/installing-workflow/configuring-postgres.md]: That's a good question. I think this is the classic conundrum that we have created for ourselves by generating our docs from master / _about_ what is in master. It raises this question every time a new feature is introduced and docs relating to that feature (install docs, for instance) must mention a version number.

Can we LGTM and merge deis/charts#286 and then just sit on this PR for now?

File: src/installing-workflow/configuring-postgres.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# Configuring Postgres
+
+Deis Workflow's controller component relies on a PostgreSQL database to store platform state.
+
+By default, Deis Workflow ships with the [database] component, which provides an in-cluster PostgreSQL database backed up to in-cluster or off-cluster [object storage]. Currently, for object storage, which is utilized by _several_ Workflow components, only off-cluster solutions such as S3 of GCS are recommended in production environments. Experience has shown that many operators already opting for off-cluster object storage similarly prefer to host Postgres off-cluster as well, using Amazon RDS or similar. When excercising both options, a Workflow installation becomes entirely stateless, and is thus restored or rebuilt with greater ease if the need should ever arise.
```
Comment (arschles) [inline src/installing-workflow/configuring-postgres.md]: nit: can you shorten this a tiny by saying "... thus restored or rebuilt with greater ease should the need ever arise."

File: src/installing-workflow/configuring-postgres.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        * Update the values in the `[database]` configuration section to properly reflect all connection details.
+        * Save your changes.
+    * Note: Whether using environment variables or `tpl/generate_params.toml`, you do not need to (and must not) base64 encode any values, as the Helm Classic chart will automatically handle encoding as necessary.
+* **Step 3:** Re-generate the Helm Classic chart by running `helmc generate -x manifests workflow-v2.1.0`
+* **Step 4:** Check the generated files in your `manifests` directory. You should see:
+    * `deis-controller-rc.yaml` contains relevant connection details.
+    * `deis-database-secret-creds.yaml` exists and contains base64 encoded database username and password.
+    * No other database-related Kubernetes resources are defined. i.e. none of `database-database-service-account.yaml`, `database-database-service.yaml`, or `database-database-rc.yaml` exist.
+
+You are now ready to `helmc install workflow-v2.1.0` as usual.
```
Comment (arschles) [inline src/installing-workflow/configuring-postgres.md]: link here to the deis workflow install page?

File: src/installing-workflow/configuring-postgres.md (hunk 2 above)
Comment (krancour) [inline src/installing-workflow/configuring-postgres.md]: Sure.

File: src/installing-workflow/configuring-postgres.md (hunk 3 above)
Comment (krancour) [inline src/installing-workflow/configuring-postgres.md]: Good idea. I'll add.

**Comment by bacongobbler** [comment]
I think this is safe to merge now that we are going through the release process and going to be releasing a v2.1.0 chart hopefully today, but it's your call @krancour. :)

**Pull request merged**